File: TpWeb-main/back/src/main/java/com/research/paper/service/paper/CollectionService.java
package com.research.paper.service.paper;

import com.research.paper.dto.request.paper.collection.CollectionCreationRequest;
import com.research.paper.dto.request.paper.collection.CollectionUpdateRequest;
import com.research.paper.dto.response.paper.CollectionResponse;

import java.util.List;

public interface CollectionService {
    void addCollection(CollectionCreationRequest collectionCreationRequest);
    void updateCollection(CollectionUpdateRequest collectionUpdateRequest , String collectionId);
    void deleteCollection(String collectionId);
    List<CollectionResponse> getAllCollectionsByUserId(String userId);
    CollectionResponse getCollectionById(String collectionId);
}

File: TpWeb-main/angular/scholarIA/src/app/services/collection.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface CollectionCreationRequest {
  name: string;
  description?: string;
  [key: string]: any;
}

export interface CollectionUpdateRequest {
  name?: string;
  description?: string;
  [key: string]: any;
}

export interface CollectionResponse {
  id: string;
  name: string;
  description?: string;
  userId?: string;
  papers?: string[];
  createdAt?: string;
  updatedAt?: string;
  [key: string]: any;
}

@Injectable({
  providedIn: 'root'
})
export class CollectionService {
  private apiUrl = 'http://localhost:8080/api/v1/collection';

  constructor(private http: HttpClient) { }

  /**
   * Add a new collection
   */
  addCollection(request: CollectionCreationRequest): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/add`, request);
  }

  /**
   * Update a collection
   */
  updateCollection(collectionId: string, request: CollectionUpdateRequest): Observable<void> {
    return this.http.patch<void>(`${this.apiUrl}/${collectionId}`, request);
  }

  /**
   * Get all collections for current user
   */
  getAllCollectionsByUserId(): Observable<CollectionResponse[]> {
    return this.http.get<CollectionResponse[]>(`${this.apiUrl}/user`);
  }

  /**
   * Get collection by ID
   */
  getCollectionById(collectionId: string): Observable<CollectionResponse> {
    return this.http.get<CollectionResponse>(`${this.apiUrl}/${collectionId}`);
  }
}
